src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java

```diff
@@ -0,0 +1,65 @@
+package com.inqoo.trainingservice.app.customer;
+
+import org.apache.commons.validator.routines.EmailValidator;
+import org.springframework.stereotype.Service;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Service
+public class CustomerService {
+    private final CustomerRepository customerRepository;
+
+    public CustomerService(CustomerRepository customerRepository) {
+        this.customerRepository = customerRepository;
+    }
+
+    private Pattern regexPatternMobile;
+    private Pattern regexPatternHomeNumber;
+    private Matcher regMatcher;
+
+    public Customer saveNewCustomer(Customer customer) {
+        if (!validateEmailAddress(customer.getEmailAddress())) {
+            throw new EmailNotValidException("Mail not valid");
```

Comment:
Those messages can be moved into exception directly

Reply:
done